client/utils/dagreAutoLayout.ts

```diff
@@ -0,0 +1,53 @@
+import dagre from '@dagrejs/dagre';
+// Placeholder for initial nodes and edges types
+import type { Node, Edge } from 'reactflow';
+
+const setupDagreFlow = () => {
+  const dagreFlow = new dagre.graphlib.Graph();
+  dagreFlow.setDefaultEdgeLabel(() => ({}));
+  dagreFlow.setGraph({ rankdir: 'TB' });
+  return dagreFlow;
+};
+
+const nodeWidth = 168;
+const nodeHeight = 44;
+
+function dagreAutoLayout(initialNodes: Node[], initialEdges: Edge[]) {
+  const dagreGraph = setupDagreFlow();
+
+  initialNodes.forEach((node) => {
+    dagreGraph.setNode(node.id, { width: nodeWidth, height: nodeHeight });
+  });
+
+  initialEdges.forEach((edge) => {
+    dagreGraph.setEdge(edge.source, edge.target);
+  });
+
+  //already spreaded out the nodes
+  dagre.layout(dagreGraph);
+
+  initialNodes.forEach((node) => {
```

Comment:
I wonder if mutating the initialNodes array would cause any unexpected side effects in React.🤔